engine/execution/ingestion/uploader/uploader.go

```diff
@@ -26,74 +27,99 @@ func NewAsyncUploader(uploader Uploader,
 	maxRetryNumber uint64,
 	log zerolog.Logger,
 	metrics module.ExecutionMetrics) *AsyncUploader {
-	return &AsyncUploader{
-		unit:                engine.NewUnit(),
+	a := &AsyncUploader{
 		uploader:            uploader,
 		log:                 log.With().Str("component", "block_data_uploader").Logger(),
 		metrics:             metrics,
 		retryInitialTimeout: retryInitialTimeout,
 		maxRetryNumber:      maxRetryNumber,
+		queue:               make(chan *execution.ComputationResult, 100),
 	}
+	builder := component.NewComponentManagerBuilder()
+	for i := 0; i < 3; i++ {
```

Comment:
I'd be inclined to increase the number of workers here, again to ensure we can work through the backlog. Goroutines are very cheap. Maybe 10 workers?